components/benefit_sponsors/app/models/benefit_sponsors/members/member_group.rb

```diff
@@ -62,7 +62,8 @@ def each
       end
     end
 
-    def clone_for_coverage(new_product)
+    def clone_for_coverage(new_product, action = nil)
+      group_enrollment.previous_product = new_product if action.to_s == 'calc_childcare_subsidy'
```

Comment:
is there any risk by setting `previous_product` irrespective of action ? (I mean, can we avoid introducing `action` here ?)

Reply:
we cannot set previous_product on different actions, the purpose of adding action params to this method is to resolve edge case scenario of the ticket, this method will be called on different scenarios but when we are updating childcare_subsidy to an enrollment then it is considering eligibility_date instead coverage start on because it is getting slcsp product which should be avoided. @nisanthyaganti9